### ISSUE TITLE
addons.mozilla.org rmb downloads dont work

### ISSUE THREAD

**Issue by dezhavu** [uBlockOrigin/uMatrix-issues#91]
### Prerequisites

- [x ] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported.
    - [x ] I also searched the existing issues at <https://github.com/gorhill/uMatrix/issues>
- [x ] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uMatrix](https://old.reddit.com/r/uMatrix/)
- I tried to reproduce the issue when...
    - [x ] uMatrix is the only extension
    - [x ] uMatrix with default lists/settings
    - [x ] using a new, unmodified browser profile
- [x ] I am running the latest version of uMatrix
- [x ] I checked the [documentation](https://github.com/gorhill/uMatrix/wiki) to understand that the issue I report is not a normal behavior
- [x ] I used the logger to rule out that the issue is caused by my ruleset

### Steps to Reproduce

ok, this one came up suddenly:

1. go to addons.mozilla.org and choose any addon:
https://addons.mozilla.org/en-US/firefox/addon/multi-account-containers
2. RIGHT CLICK on green button "Add to Firefox" > Save Link As... > and you get alert:
"The download cannot be saved because an unknown error occurred.
Please try again."

### Ruleset
defaults

https-strict: behind-the-scene false
matrix-off: about-scheme true
matrix-off: behind-the-scene true
matrix-off: chrome-extension-scheme true
matrix-off: chrome-scheme true
matrix-off: moz-extension-scheme true
matrix-off: opera-scheme true
matrix-off: wyciwyg-scheme true
noscript-spoof: * true
referrer-spoof: * true
referrer-spoof: behind-the-scene false
* * * block
* * css allow
* * frame block
* * image allow
* 1st-party * allow
* 1st-party frame allow

### Supporting evidence
<a href="http://www.imagebam.com/image/0fc4cd1071253524" target="_blank"><img src="http://thumbs2.imagebam.com/4b/72/1d/0fc4cd1071253524.jpg" alt="imagebam.com"></a> 

### Your environment

* uMatrix version: latest
* Browser Name and version: ffx 64 x64
* Operating System and version: win10 1809 x64

Firefox 64 all default settings+ these custom two to prevent addons.mozilla.org blocks:
privacy.resistFingerprinting.block_mozAddonManager=true
( btw privacy.resistFingerprinting=false )

**Comment by uBlock-user** [comment]
@gorhill That blocked number count doesn't appear on other cell column in the UI matrix on v1.13.15.101

**Comment by uBlock-user** [comment]
@gwarser That doesn't happen in rc0 build, so that's not the case. This is a regression.

**Comment by uBlock-user** [comment]
so is it a regression or not a regression ?

**Comment by uBlock-user** [comment]
>Whatever I could reproduce given the missing information, I didn't see anything unexpected.

You mean OP's case ?

**Comment by gorhill** [comment]
As per logger, I got one behind-the-scene network request on my side, everything else was properly reported in the matrix. Screenshots of popup panel are pointless if they are not accompanied by logger output if trying to make a point something is wrong on the panel. I will investigate if and only if someone makes the case that something in the logger, which not tabless, is not reported in the panel.

**Comment by gorhill** [comment]
The missing information by OP is the value of his `extensions.webextensions.restrictedDomains` string, so I just removed `addons.mozilla.org` on my side.

**Comment by uBlock-user** [comment]
> Screenshots of popup panel are pointless if they are not accompanied by logger output if trying to make a point something is wrong in the panel.

I didn't post any screenshots here, not sure whether you're addressing this to me or to the OP.

**Comment by Remu-rin** [comment]
@gorhill 
> As per logger, I got one behind-the-scene network request on my side, everything else was properly reported in the matrix.

Previously, starting from 1.3.8 (see https://github.com/uBlockOrigin/uMatrix-issues/issues/5#issuecomment-385242250) and up to 1.3.14 (or 1.13.15rc0, as stated above by the users of dev builds) uMatrix also reported "save as" requests in the popup (on github releases page for example, or AMO if you remove restriction for extensions to work). It was easier to see and unblock for users.
> In doing so, it also tries to find the tab which matches best that context. So when you try to "Save as File...", you will have a network request of type "other" reported in the matrix.

Now in 1.3.16 it is broken and only visible in the logger. Hence this issue, #92 and partly #62. Is this change intentional or a regression?

**Comment by gorhill** [comment]
Before 1.3.16 there was [a patch](https://github.com/gorhill/uMatrix/commit/98e012599c30dfada7e779b4eb9c9bc874ef34e7#diff-e89c0f548b51d6b02baea1e832b9d8ba) to map a tab-less network request to an existing tab where possible. This patch of code was put in place in order to allow users to be able to set rules on these tab-less network requests for which a context was available, as there was no other way to set rules for these requests other than blindly guessing and manually entering them in the rule editor.

With 1.3.16, it is now possible to properly set rules with all the proper context information, _even_ for tab-less network requests, such that the patch is no longer needed.

That patch was not the proper solution, it was added strictly to have at least some ways to set rules for tab-less requests until the proper solution was provided, thus the patch was removed from 1.3.16.

It is not the proper solution because uMatrix merely looked-up a tab for which a tab-less request was a match, and the resulting tab may or may not have been directly related to the request. If a site uses a service worker and you have 10 tabs opened for that site, uMatrix would have _maybe_ be able to find a tab to which to report it, and the tab may or may not be the one in which an action was triggered to cause the tab-less request to be fired.

The proper way is to report the network requests as accurately as possible, and thus tab-less network requests can't be reported in tab-bound popup panel.

**Comment by uBlock-user** [comment]
> thus tab-less network requests can't be reported in tab-bound popup panel

So [gwarser's suggestion](https://github.com/uBlockOrigin/uMatrix-issues/issues/62#issuecomment-449847012) can't be implemented ?
